File: nim/sieve.nim
import sequtils, strutils, math

proc sieve (maxnum: int): seq[int] =
  var nums = newSeq[uint8](maxnum + 1)
  for i in countup(3, maxnum, 2):
    nums[i] = 1
    
  for i in countup(3, int(ceil(sqrt(float(maxnum)))), 2):
    if nums[i] == 0:
      continue
      
    for j in countup( i * 3, maxnum, i * 2):
      if nums[j] == 1:
        nums[j] = 0

  result = @[2]
  for i in countup(3, maxnum, 2):
    if nums[i] == 1:
      result.add(i)
      
let primes = sieve(readLine(stdin).parseInt)
echo( primes.len, " ", primes[primes.len - 1] )

File: python/sieve.py
import math

def sieve(n):
    nums = list(range(n + 1))
    for i in range( 3, math.ceil( n**0.5 ), 2):
        if nums[i] == 0:
            continue
        for j in range( i * i, n, i * 2 ):
            if nums[j] != 0:
                nums[j] = 0

    return [2] + list( filter( lambda x: x % 2 != 0 and x > 2, nums ) )

primes = sieve( int(input()) )
print( '{} {}'.format(len(primes), primes[-1]) )
